legend-engine-xts-data-space/legend-engine-xt-data-space-generation/src/main/java/org/finos/legend/engine/generation/analytics/DataSpaceAnalyticsHelper.java

```diff
@@ -124,6 +124,107 @@ private static DataSpaceExecutableResult buildExecutableResult(ResultType result
         return null;
     }
 
+    private static List<DataSpaceExecutableAnalysisInfo> buildDataSpaceExecutableAnalysisInfo(PureModel pureModel, DataSpace dataSpaceProtocol, PureModelContextData pureModelContextData)
+    {
+        List<DataSpaceExecutableAnalysisInfo> dataSpaceExecutableAnalysisInfos = null;
+        Set<String> qualifiedMappingPath = dataSpaceProtocol.executionContexts.stream().map(e -> e.mapping.path).collect(Collectors.toSet());
+        Set<String> qualifiedRuntimePath = dataSpaceProtocol.executionContexts.stream().map(e -> e.defaultRuntime.path).collect(Collectors.toSet());
+        if (dataSpaceProtocol.executables != null)
+        {
+            dataSpaceExecutableAnalysisInfos = dataSpaceProtocol.executables.stream()
+                    .filter(ex -> ex instanceof DataSpacePackageableElementExecutable)
+                    .map(executable ->
+                    {
+                        String path = ((DataSpacePackageableElementExecutable) executable).executable.path;
+                        try
+                        {
+                            PackageableElement service = pureModel.getPackageableElement(path);
+                            Optional<org.finos.legend.engine.protocol.pure.v1.model.packageableElement.PackageableElement> v1Service =  pureModelContextData.getElements().stream().filter(e -> e.getPath().equals(path)).findAny();
+                            if (v1Service.isPresent() && v1Service.get() instanceof Service)
+                            {
+                                if (service instanceof Root_meta_legend_service_metamodel_Service)
```

Comment:
looking at the metamodel seems to make this a bit hard to read. could we just read the v1Service properties. 
Most of them are 1-1.

Reply:
I found it's hard to get the runtime from  public Runtime runtime;